src/cargo/ops/cargo_compile.rs

```diff
@@ -582,6 +590,82 @@ pub fn create_bcx<'a, 'cfg>(
         }
     }
 
+    if rustdoc_scrape_examples && units.len() > 0 {
+        let mut args = Vec::new();
+
+        let get_repository = || -> Option<String> {
+            let repository = units[0].pkg.manifest().metadata().repository.as_ref()?;
+            let repo_url = Url::parse(repository).ok()?;
+            let domain = repo_url.domain()?;
+
+            match domain {
+                "github.com" => {
+                    // TODO(wcrichto): is there a canonical way to get the current commit?
+                    // Would prefer that over linking to master
+                    let gitref = "master";
+                    Some(format!("{}/tree/{}", String::from(repo_url), gitref))
+                }
+                _ => None,
+            }
+        };
+
+        if let Some(repository) = get_repository() {
+            args.push("--repository-url".to_owned());
+            args.push(repository);
+        }
+
+        // Needed while --scrape-examples and --repository-url are unstable
+        args.push("-Zunstable-options".to_owned());
+
+        // Get examples via the `--examples` filter
+        let mut example_compile_opts = CompileOptions::new(ws.config(), CompileMode::Doctest)?;
+        example_compile_opts.cli_features = options.cli_features.clone();
+        example_compile_opts.build_config.mode = CompileMode::Doctest;
+        example_compile_opts.spec = Packages::All;
+        example_compile_opts.filter = CompileFilter::Only {
+            all_targets: false,
+            lib: LibRule::False,
+            bins: FilterRule::none(),
+            examples: FilterRule::All,
+            benches: FilterRule::none(),
+            tests: FilterRule::none(),
+        };
+        let example_compilation = ops::compile(ws, &example_compile_opts)?;
+
+        // FIXME(wcrichto): ideally the call locations would be cached in the target/ directory
+        // so they don't have to be recomputed
+        let td = TempFileBuilder::new().prefix("cargo-doc").tempdir()?;
+
+        for (i, doc_test) in example_compilation.to_doc_test.iter().enumerate() {
+            let mut p = doc_test.rustdoc_process(&example_compilation)?;
+            let src = doc_test.unit.target.src_path().path().unwrap();
+            p.arg(src);
+
+            let path = td.path().join(format!("{}.json", i));
+            p.arg("--scrape-examples").arg(&path);
+            p.arg("--workspace-root")
+                .arg(&ws.root().display().to_string());
+            p.arg("-Z").arg("unstable-options");
+            config
+                .shell()
+                .verbose(|shell| shell.status("Running", p.to_string()))?;
+            p.exec()?;
+            args.extend_from_slice(&["--with-examples".to_string(), format!("{}", path.display())]);
```

Comment:
to_display is wrong, it doesn't handle non-unicode filepaths. you may need to change extra_compiler_args to be an OsStr or something.

Reply:
Fixed.